### PR TITLE
use common function to parse arg + generic format for test case error

### PR DESCRIPTION
**Pull request by s8sg**
Signed-off-by: Swarvanu Sengupta <swarvanusg@gmail.com>

## Description

- Use common function `parseMap` in `commands/deploy.go` to parse `build-arg`
- Change of test cases accordingly
- Fix govet issue related with missing parameter:
  `Fatal call has possible formatting directive %#v`
  `Fatalf no formatting directive had passed`

## Motivation and Context
This is a followup for the issue: [#363](https://github.com/openfaas/faas-cli/issues/363)
and PR: [#364](https://github.com/openfaas/faas-cli/pull/364) 
- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
Testing have been done manually with the provided Dockerfile 
Dockerfile
```dockerfile
rm stack.yml # no conflicts
faas new --lang dockerfile argstest

cat > ./argstest/Dockerfile
FROM alpine:3.6
# 1. Use any image as your base image, or "scratch"
# 2. Add fwatchdog binary via https://github.com/openfaas/faas/releases/
# 3. Then set fprocess to the process you want to invoke per request - i.e. "cat" or"my_binary"

#ADD https://github.com/openfaas/faas/releases/download/0.7.8/fwatchdog /usr/bin
#RUN chmod +x /usr/bin/fwatchdog

RUN apk --no-cache add curl \
    && echo "Pulling watchdog binary from Github." \
    && curl -sSL https://github.com/openfaas/faas/releases/download/0.7.8/fwatchdog > /usr/bin/fwatchdog \
    && chmod +x /usr/bin/fwatchdog \
    && apk del curl --no-cache

ARG CAPTURE_THIS

RUN echo ${CAPTURE_THIS} | tee /captured.txt
ENV fprocess="cat /captured.txt"
# Set to true to see request in function logs
ENV write_debug="false"

HEALTHCHECK --interval=5s CMD [ -e /tmp/.lock ] || exit 1
CMD [ "fwatchdog" ]
```
Build:
```bash
go build && ./faas-cli build -f ./argtest.yml --build-arg CAPTURE_THIS=my-value --no-cache
```
Check the data was populated:
```bash
docker run -ti argtest cat /captured.txt
my-value
```

## Types of changes
PR followup

## Checklist:

- [x] My code follows the code style of this project.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: commands/build_test.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func Test_parseBuildArgs_EmptyKey(t *testing.T) {
 }
 
 func Test_parseBuildArgs_MultipleSeparators(t *testing.T) {
```
Comment (alexellis) [inline commands/build_test.go]: @s8sg the test name should change from parseBuildArgs to `parseMap` in this instance right?

File: commands/build_test.go [hunk 2]
```diff
@@ -39,19 +39,19 @@ func Test_parseBuildArgs_ValidParts(t *testing.T) {
...
+	want := "label format is not correct, needs key=value"
 	if err != nil && err.Error() != want {
 		t.Errorf("Expected an error due to missing seperator")
 		t.Fail()
 	}
 }
 
 func Test_parseBuildArgs_EmptyKey(t *testing.T) {
-	_, err := parseBuildArgs([]string{"=v"})
+	_, err := parseMap([]string{"=v"}, "build-arg")
```
Comment (alexellis) [inline commands/build_test.go]: Are there existing tests covering `parseMap` or are these the only ones? If they are the only ones then we should rename the test file too.

File: commands/build.go [hunk 3]
```diff
@@ -154,8 +125,10 @@ func runBuild(cmd *cobra.Command, args []string) error {
 
 func build(services *stack.Services, queueDepth int, shrinkwrap bool) {
 	wg := sync.WaitGroup{}
+	defer wg.Wait()
```
Comment (alexellis) [inline commands/build.go]: Looks like a smart move 👍

File: commands/build_test.go (hunk 1 above)
Comment (s8sg) [reply commands/build_test.go]: @alexellis Right. I'll update

File: commands/build_test.go (hunk 2 above)
Comment (s8sg) [reply commands/build_test.go]: these the only ones. How about creating a `utils.go` under `commands` package for  `parseMap`and other common utils.  We can separately have  test cases for `utils`

**Comment by alexellis** [comment]
I really dislike names like utils and common. The test name could reflect the method we are testing even that would be an improvement.

File: proxy/list_test.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ func Test_ListFunctions(t *testing.T) {
 	}
 	for k, v := range result {
 		if expectedListFunctionsResponse[k] != v {
-			t.Fatal("Expeceted: %#v - Actual: %#v", expectedListFunctionsResponse[k], v)
+			t.Fatalf("Expeceted: %#v - Actual: %#v", expectedListFunctionsResponse[k], v)
```
Comment (alexellis) [inline proxy/list_test.go]: We have a typo here. Also the Go test output should be want and got rather than expected actual. Can you update please?

**Comment by s8sg** [comment]
I'm getting the below issue:
```
$ faas-cli new myFunc --lang go
$ faas-cli build -f myFunc.yml
Failed to parse build-arg: <nil>
```
Please hold the PR

**Pull request closed without merging**